tests/core/test_test.py

```diff
@@ -191,17 +251,22 @@ async def test_action_unlikely_intent_2(
     # We train on the above story so that RulePolicy can memorize
     # it and we don't have to worry about other actions being
     # predicted correctly.
-    agent = await _train_rule_based_agent(moodbot_domain, file_name)
+    agent = await _train_rule_based_agent(
+        moodbot_domain, file_name, monkeypatch, ignore_action_unlikely_intent=False
+    )
 
     result = await rasa.core.test.test(
-        str(file_name), agent, out_directory=str(tmp_path),
+        str(file_name),
+        agent,
+        out_directory=str(tmp_path),
+        fail_on_prediction_errors=True,
     )
     assert "report" in result.keys()
     assert result["report"]["conversation_accuracy"]["correct"] == 1
     assert result["report"]["conversation_accuracy"]["with_warnings"] == 0
 
 
-async def test_action_unlikely_intent_complete(
+async def test_action_unlikely_intent_fail_on_prediction_errors(
```

Comment:
I think from what you explained as the utility of this test, it looks like we should rename this test to not indicate that this is specific to `action_unlikely_intent` as we expect it to raise this exception on any failed prediction except `action_unlikely_intent`.